services/requester/local_client.go

```diff
@@ -0,0 +1,100 @@
+package requester
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/onflow/go-ethereum/common"
+
+	"github.com/onflow/flow-evm-gateway/services/state"
+	"github.com/onflow/flow-evm-gateway/storage"
+)
+
+var _ EVMClient = &LocalClient{}
+
+func NewLocalClient(state *state.BlockState, blocks storage.BlockIndexer) *LocalClient {
+	return &LocalClient{
+		state:  state,
+		blocks: blocks,
+	}
+}
+
+type LocalClient struct {
+	state  *state.BlockState
+	blocks storage.BlockIndexer
+}
+
+func (l *LocalClient) SendRawTransaction(
+	ctx context.Context,
+	data []byte,
+) (common.Hash, error) {
+	return common.Hash{}, fmt.Errorf("local client is read-only")
+}
+
+func (l *LocalClient) GetBalance(
+	ctx context.Context,
+	address common.Address,
+	evmHeight int64,
+) (*big.Int, error) {
+	bal := l.state.GetBalance(address)
```

Comment:
How do we know that the `*state.BlockState` matches the given `evmHeight` :thinking: ? The same goes for all methods that accept historical blocks.